src/main/java/io/shiftleft/tarpit/FileUploader.java

```diff
@@ -0,0 +1,69 @@
+package io.shiftleft.tarpit;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+import io.shiftleft.tarpit.util.Unzipper;
+
+/**
+ * Servlet implementation class FileUploader
+ */
+@WebServlet("/FileUploader")
+@MultipartConfig
+public class FileUploader extends HttpServlet {
+
+  private static final long serialVersionUID = 1L;
+  private static String productSourceFolder = System.getenv("PRODUCT_SRC_FOLDER");
+  private static String productDestinationFolder = System.getenv("PRODUCT_DST_FOLDER");
+
+  /**
+   * @see HttpServlet#HttpServlet()
+   */
+  public FileUploader() {
+    super();
+  }
+
+
+  /**
+   * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+   */
+  protected void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+
+    Part filePart = request.getPart("zipFile");
+
+    InputStream input = filePart.getInputStream();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *input*. The resource is referenced in statements at the following line: 56. The resource closure statement is at line: 60. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by InputStream.read. Either a) close *input* in a try-finally block or b) close the resource by declaring *input* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).